Library/Homebrew/rubocops/bottle_block_cop.rb

```diff
@@ -1,20 +1,31 @@
+require_relative "./extend/formula_cop"
+
 module RuboCop
   module Cop
     module Homebrew
-      class CorrectBottleBlock < Cop
-        MSG = "Use rebuild instead of revision in bottle block".freeze
+      # This cop audits `bottle` block in Formulae
+      #
+      # - `rebuild` should be used instead of `revision` in `bottle` block
 
-        def on_block(node)
-          return if block_length(node).zero?
-          method, _args, body = *node
-          _keyword, method_name = *method
+      class CorrectBottleBlock < FormulaCop
+        MSG = "Use rebuild instead of revision in bottle block".freeze
 
-          return unless method_name == :bottle
-          check_revision?(body)
+        def audit_formula(_node, _class_node, _parent_class_node, formula_class_body_node)
+          check(formula_class_body_node)
```

Comment:
Ideally this method would contain code like the following:
```ruby
def audit_formula
  bottle = find_block(:bottle)
  return unless bottle
  if method_called?(bottle, :revision)
    problem "Use rebuild instead of revision in a bottle block"
  end
end
```

The `find_block` and `method_called?` methods should be defined in `formula_cop.rb` and the various instance variables set by `on_class` or something it calls.